src/controllers/group.js

```diff
@@ -112,6 +115,53 @@ const Group = {
 
     return res.status(204).send();
   },
+
+  upload: async (req, res, next) => {
+    const { id, userId } = req.params;
+    let result;
+
+    try {
+      result = await GroupModel.findById(id).exec();
+    } catch (err) {
+      return next(new APIError(
+        'Group could not find a group with ID',
+        'Invalid ID provided',
+        404,
+        `/groups/${id}`,
+      ));
+    }
+    if (!req.file) {
+      return next(new APIError(
+        'Group Could not upload file',
+        'No file provided',
+        500,
+        `/groups/${id}`,
+      ));
+    }
+
+    const imageBuffer = await sharp(req.file.buffer)
+      .jpeg()
+      .toBuffer();
+    const fileName = `${uuidv4()}.jpeg`;
+    const key = `groups/${id}/${fileName}`;
+
+    try {
+      await S3.uploadObject(key, imageBuffer);
+    } catch (err) {
+      return next(new APIError());
+    }
+
+    const image = new ImageModel(
+      fileName, userId, new Date(),
+    );
+
+    result = await GroupModel.update(
+      { _id: result.id },
+      { $push: { images: image } },
+    );
+
+    return res.status(200).send(result.toJSON());
```

Comment:
`result` is a query object, which does not have a `.toJSON()` method.

Reply:
I don't think we need to return any data to the response I would just have 204 response

Reply:
I went ahead an made the change